src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -194,7 +195,8 @@ public PasswordParameterEnvironmentExpander(Run<?, ?> run) {
             ParametersAction action = run.getAction(ParametersAction.class);
             if (action != null) {
                 passwordParameterVariables = action.getParameters().stream()
-                        .filter(PasswordParameterValue.class::isInstance)
+                        .filter(e -> PasswordParameterValue.class.isInstance(e)
+                                        && !((Secret) e.getValue()).getPlainText().isEmpty())
```

Comment:
I was worried about possible NPEs here but it looks like `PasswordParameterValue.getValue` and `Secret.getPlainText` are both annotated with `@NonNull` so it should be fine.